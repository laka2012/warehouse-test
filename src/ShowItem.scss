.showitem{
    display: inline-block;
    padding: 15px;
    margin: 5px;
    border: 1px solid #b4bac4;
    .name {
        border-bottom: 1px solid #b4bac4;
        padding-bottom: 5px;
        text-align: center;
        width: 300px;
    }
    .image {
        margin: 10px 0px;
        img {
            width: 300px;
        }
    }
    .bottom {
        border-top: 1px solid #b4bac4;
        padding: 5px;
        .price {
            float: left;
            margin-left: 20px;
        }
        .add-cart {
            float: right;
            margin-right: 20px;
        }
    }
    &:hover {
        box-shadow: 0 0 20px #a4c2f2; 
    }
}