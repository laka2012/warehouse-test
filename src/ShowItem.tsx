import React, { Component } from 'react';
import './ShowItem.scss';
import Item from './IItem';

export interface Props {
	item: Item;
	addToCart: (item: Item) => void; 
}

class ShowItem extends Component<Props> {
	constructor(props: Props) {
		super(props);
		this.state = {isToggleOn: true};
		this.handleAddCartClick = this.handleAddCartClick.bind(this);
	}

	handleAddCartClick() {
		this.props.addToCart(this.props.item);
	}
	
	render() {
		return (
			<div className='showitem'>
				<div className='name'>{this.props.item.name}</div>
				<div className='image'>
					<img src={this.props.item.image_url} ></img>
				</div>
				<div className='bottom'>
					<div className='price'>${this.props.item.price}</div>
					<button className='add-cart' onClick={this.handleAddCartClick}>Add To Cart</button>
				</div>
			</div>
		);
	}
}

export default ShowItem;
