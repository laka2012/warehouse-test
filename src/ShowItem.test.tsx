import React from 'react';
import { shallow } from 'enzyme';
import ShowItem, {Props} from './ShowItem';
import Item from './IItem';

function showItemTest(props: Partial<Props> = {}) {
    return (
        <ShowItem {...{
            item: {
                date_added: 'test',
                type: 'test',
                sku: 'test',
                group_id: 'test',
                name: 'test',
                model: 'test',
                brand: 'test',
                url: 'test',
                image_url: 'test',
                on_special: true,
                price: 100,
                flybuys_points: 1005,
                shipping_class: 'test',
                add_to_cart: true,
                categories: 'test',
                in_stock: true,
                sold_out: false
            },
	        addToCart: (item: Item) => {},
            ...props}}>
        </ShowItem>
    );
}

describe('ShowItem', () => {
	it('smoke test', () => {
		const wrapper = shallow(
			showItemTest()
		);
		expect(wrapper).toMatchSnapshot();
    });
});