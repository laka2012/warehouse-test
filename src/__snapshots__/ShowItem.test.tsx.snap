// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowItem smoke test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShowItem
    addToCart={[Function]}
    item={
      Object {
        "add_to_cart": true,
        "brand": "test",
        "categories": "test",
        "date_added": "test",
        "flybuys_points": 1005,
        "group_id": "test",
        "image_url": "test",
        "in_stock": true,
        "model": "test",
        "name": "test",
        "on_special": true,
        "price": 100,
        "shipping_class": "test",
        "sku": "test",
        "sold_out": false,
        "type": "test",
        "url": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="name"
        >
          test
        </div>,
        <div
          className="image"
        >
          <img
            src="test"
          />
        </div>,
        <div
          className="bottom"
        >
          <div
            className="price"
          >
            $
            100
          </div>
          <button
            className="add-cart"
            onClick={[Function]}
          >
            Add To Cart
          </button>
        </div>,
      ],
      "className": "showitem",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "name",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            src="test"
          />,
          "className": "image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "test",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="price"
            >
              $
              100
            </div>,
            <button
              className="add-cart"
              onClick={[Function]}
            >
              Add To Cart
            </button>,
          ],
          "className": "bottom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "$",
                100,
              ],
              "className": "price",
            },
            "ref": null,
            "rendered": Array [
              "$",
              100,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add To Cart",
              "className": "add-cart",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add To Cart",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="name"
          >
            test
          </div>,
          <div
            className="image"
          >
            <img
              src="test"
            />
          </div>,
          <div
            className="bottom"
          >
            <div
              className="price"
            >
              $
              100
            </div>
            <button
              className="add-cart"
              onClick={[Function]}
            >
              Add To Cart
            </button>
          </div>,
        ],
        "className": "showitem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": "name",
          },
          "ref": null,
          "rendered": "test",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="test"
            />,
            "className": "image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "test",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="price"
              >
                $
                100
              </div>,
              <button
                className="add-cart"
                onClick={[Function]}
              >
                Add To Cart
              </button>,
            ],
            "className": "bottom",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "$",
                  100,
                ],
                "className": "price",
              },
              "ref": null,
              "rendered": Array [
                "$",
                100,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add To Cart",
                "className": "add-cart",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add To Cart",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
